src/text-editor-component.js

```diff
@@ -3586,7 +3587,7 @@ class CursorsAndInputComponent {
         compositionupdate: didCompositionUpdate,
         compositionend: didCompositionEnd
       },
-      tabIndex: -1,
+      tabIndex: this.tabIndex,
```

Comment:
I think this is an attempt to access the [`tabIndex` property on the `TextEditorComponent` object](https://github.com/atom/atom/blob/df2f3c3124a6e7ea7f9e23c912cff9abfefda0b9/src/text-editor-component.js#L173). However, we're inside the [`CursorsAndInputComponent` class](https://github.com/atom/atom/blob/df2f3c3124a6e7ea7f9e23c912cff9abfefda0b9/src/text-editor-component.js#L3494-L3604) here, and the `CursorsAndInputComponent` objects don't currently have a `tabIndex` property. As a result, `this.tabIndex` is always `undefined`, and that results in the `input` element getting its tabIndex set to `0`. I _think_ this is the cause of the issues you noted in the [possible drawbacks section](https://github.com/atom/atom/pull/16127#issue-151324878).

I've attempted to resolve those issues in #16492. If you have a chance to look at that pull request, I'd love to have your feedback. :bow: